### PR TITLE
support adding model's weights to GPTQ loss

### PR DESCRIPTION
**Pull request by elad-c**
Add float and quantized kernels to GPTQ loss function API.
Changed get_trainble_parameters to return the trainable parameters as a list of lists, each containing a layer's weights for trainin, and 2 mote lists containing the quantized and float kernels of the layers to be passed to the GPTQ loss function.

### CODE REVIEW

File: model_compression_toolkit/keras/gradient_ptq/training_wrapper.py [hunk 1]
```diff
@@ -69,9 +69,15 @@ def gptq_training_wrapper(tg: Graph,
...
 
-    trainable_weights = get_trainable_parameters(fxp_model,
-                                                 fw_info,
-                                                 add_bias=gptq_config.train_bias)
+    trainable_weights, flp_weights_list, fxp_weights_list = \
+        get_trainable_parameters(fxp_model,
+                                 fw_info,
+                                 add_bias=gptq_config.train_bias)
+
+    if len(compare_points) != len(trainable_weights):
```
Comment (haihabi) [inline model_compression_toolkit/keras/gradient_ptq/training_wrapper.py:77]: Where is the check that len(fxp_weights_list)=len(flp_weights_list)=len(compare_points)
Also where is the batch-normalization prior?

File: model_compression_toolkit/keras/gradient_ptq/graph_info.py [hunk 2]
```diff
@@ -48,26 +48,43 @@ def get_compare_points(input_graph: Graph) -> Tuple[List[BaseNode], List[str]]:
...
         fxp_model: Model to get its trainable parameters.
         fw_info: Framework information needed for keras kernel ops list.
         add_bias: Whether to include biases of the model (if there are) or not.
 
     Returns:
-        A list of trainable variables in a model.
+        A list of trainable variables in a model. Each item is a list of a layers weights.
+        A list of float kernels, each item is the float kernel of the layer
+        A list of quantized kernels, each item is the quantized kernel of the layer
     """
```
Comment (reuvenperetz) [inline model_compression_toolkit/keras/gradient_ptq/graph_info.py:65]: Why don't we use a separate function for this?

File: model_compression_toolkit/keras/gradient_ptq/gptq_loss.py [hunk 3]
```diff
@@ -34,13 +34,17 @@ def mse_loss(y: tf.Tensor, x: tf.Tensor, normalized: bool = True) -> tf.Tensor:
...
+                              fxp_w_list: List[List[tf.Tensor]],
+                              flp_w_list: List[List[tf.Tensor]]) -> tf.Tensor:
     """
     Compute MSE similarity between two lists of tensors
 
     Args:
         y_list: First list of tensors.
         x_list: Second list of tensors.
+        fxp_w_list: list of lists each containing a quantized model layer's trainable weights - quantized
+        flp_w_list: list of lists each containing a float model layer's weights - not quantized
```
Comment (reuvenperetz) [inline model_compression_toolkit/keras/gradient_ptq/gptq_loss.py:47]: Why don't we use them for the loss computation?

File: docs/api/api_docs/classes/GradientPTQConfig.html [hunk 4]
```diff
@@ -53,7 +53,7 @@ <h3>Navigation</h3>
 <dd class="field-odd"><ul class="simple">
 <li><p><strong>n_iter</strong> (<em>int</em>) – Number of iterations to train.</p></li>
 <li><p><strong>optimizer</strong> (<em>Any</em>) – Optimizer to use.</p></li>
-<li><p><strong>loss</strong> (<em>Callable</em>) – the loss to use. should accept 2 lists of tensors. 1st list of quantized tensors, the 2nd list is the float tensors.</p></li>
+<li><p><strong>loss</strong> (<em>Callable</em>) – The loss to use. Should accept 4 lists of tensors. 1st list of quantized tensors, the 2nd list are the float tensors, the 3rd list are the quantized weights and the 4th lsit are the float weights.</p></li>
```
Comment (reuvenperetz) [inline docs/api/api_docs/classes/GradientPTQConfig.html:56]: Do not edit HTML files. They are automatically generated before releases.
But please update the GPTQConfig and get_keras_gptq_config comments

**Review by reuvenperetz — CHANGES_REQUESTED** [review]
